HHS/Head-Start-TTADP
# [Prod] Bug fixes and accessibility improvements

## Description of change
This PR:
- Fixes the "visited" color for the back link component to have the proper color contrast
- Fixes the next step complete date validation so that a "close" enough date that passes the frontend won't error on the backend
- Fixes a bug with a materialized grant view that was causing certain merges to fail
- Properly formats the date string in the start date goals scope
- Remove the createDate filter from affecting the TTA Type graph and the roles graph on the QA dashboard

## How to test
- Confirm the link color
- Save a next step with a complete date of 11/11/2023 12:00pm and confirm that it doesn't error
- Confirm with prod data that the leech lake merge from TTAHUB-3559 works
- Visit the RTR/TTA history page and confirm that the start date and end date filters do not return an error
- Confirm that the filter on the QA dash does not change the two graphs

## Issue(s)

* https://jira.acf.gov/browse/TTAHUB-3559
* https://jira.acf.gov/browse/TTAHUB-3421
* https://jira.acf.gov/browse/TTAHUB-2295
* https://jira.acf.gov/browse/TTAHUB-3596

## Checklists

### Every PR



### Before merge to main


### Production Deploy


### After merge/deploy
